popde/density_estimate.py

```diff
@@ -351,3 +349,37 @@ def evaluate_KDEpy(self, points):
 
         return density_values
 
+
+class MultiDimRescalingBwKDEPy(VariableBwKDEPy):
+    """
+    Fit and evaluate multi-dimensional Kernel Density Estimation (KDE)
+    using KDEpy, allowing for variable per-point bandwidth and independent
+    rescaling of each dimension, i.e. a general diagonal bandwidth matrix.
+    """
+    def __init__(self, data, input_transf=None, stdize=False, rescale=None,
+                 backend='KDEpy', bandwidth=1., dim_names=None, weights=None,
+                 bandwidth_method='oned_isj')
+        """
+        bandwidth method: string
+           Name of method to get bandwidths for each dimension of the KDE
+
+        A general diagonal covariance Gaussian kernel can be written 
+          K(x - X) = const. * (h_1 h_2 ...)^(-1) exp(-(x - X)^T . diag(h_1^2, h_2^2, ...)^(-1) . (x - X) / 2)
+        This may be implemented by changing variables to w_i = x_i / h_i and 
+        using a unit kernel matrix :
+          K(w - W) = const. * exp(-(w - W)^T . diag(1, 1, ...) . (w - W) / 2).
+        More generally, the kernel in 'w' units should be proportional to the
+        unit matrix (e.g. it may vary between data points).
+        """
+        # Check compatibility of input options
+        if stdize:
+            raise ValueError("Can't standardize variables for this class!")
+        if rescale is not None:
+            raise ValueError("Can't specify rescaling for this class!")
+
+        # Use bandwidth formula to rescale dimensions
+        self.rescale = 1. / self.calc_ndim_bandwidth(bandwidth_method)
```

Comment:
this function require self.ndim which is not defined before this. I got and error   
`AttributeError: 'MultiDimRescalingBwKDEPy' object has no attribute 'ndim'`
when run.
Can we add a short example of how to call this class?